pyterrier/datasets.py

```diff
@@ -201,17 +203,51 @@ def _get_one_file(self, component, variant=None):
         return (local, filetype)
 
     def _get_all_files(self, component, variant=None, **kwargs):
-        localDir = os.path.join(self.corpus_home, component)
-        if not os.path.exists(localDir):
-            os.makedirs(localDir)
-            print("Downloading %s %s to %s" % (self.name, component, localDir))
+        if variant is None:
+            localDir = os.path.join(self.corpus_home, component)
+        else:
+            localDir = os.path.join(self.corpus_home, component, variant)
+
         kwargs = {}
         if self.user is not None:
             kwargs["auth"]=(self.user, self.password)
-        file_list = self.locations[component] if variant is None else self.locations[component][variant]
-        for (local, URL) in file_list:
+
+        direxists = os.path.exists(localDir)
+        
+        location = self.locations[component]
+        if is_lambda(location) or isinstance(location, types.FunctionType):
+            # functions are expensive to call, normally another HTTP is needed.
+            # just assume we have everthing we need if we have the local directory already
+            if direxists:
+                return localDir
```

Comment:
Should most of this function be wrapped in a try/finally to clean up the dir if anything goes wrong?

Reply:
I was wondering about touching an empty file showing that download completion has occurred, and checking if that exists.

Reply:
that sounds less ugly, yeah